Dodge.js

```diff
@@ -0,0 +1,186 @@
+import { StatusBar } from 'expo-status-bar';
+import { StyleSheet, Text, View, Dimensions, Button } from 'react-native';
+import { Knight, Circle, Brick, Grenade, MoveKnight, Shrapnel, counter} from "./Entities.js";
+import { GameEngine } from "react-native-game-engine";
+import AxisPad from 'react-native-axis-pad';
+
+//https://reactnative.dev/docs/timers.html
+//this.setTimeout??
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+let index = 0
+
+function Timer(cb, i, seconds) {
+  setTimeout(() => {
+      console.log("Timer is called!")
+      console.log(i)
+      cb(i)
+  }, seconds * 1000);
+}
+
+function DeleteEntity(i){
+delete(ents[i])
+}
+
+function Explode(i){
+  //spawns shrapnel coming from location
+  InitialShrapnel(ents[i].core.x,ents[i].core.y)
+  //delete the original projectile
+  DeleteEntity(i)
+}
+
+function Gameloop(entities, { touches }, )  {
+  //setTimeout is NOT what I want?
+  for (let id in entities) {
+    entities[id].core.update()
+  }
+  
+return entities
+}
+
+function CreateBrick(props){
+  return (
+    <View
+      style={{
+        position: 'absolute',
+        width: props.core.width,
+        height: props.core.height,
+        left: props.core.x,
+        top: props.core.y,
+        backgroundColor: props.core.backgroundColor,
+      }}
+    />
+  );
+}
+
+function CreateCircle(props){
+  return (
+
+    <View
+      style={{
+        position: 'absolute',
+        borderRadius: props.core.borderRadius,
+        width: props.core.width,
+        height: props.core.height,
+        left: props.core.x,
+        top: props.core.y,
+        backgroundColor: props.core.backgroundColor,
+      }}
+    />
+  );
+}
+
+function CreateGrenade(props){
+  return (
+    <View
+      style={{
+        position: 'absolute',
+        borderRadius: props.core.borderRadius,
+        width: props.core.width,
+        height: props.core.height,
+        left: props.core.x,
+        top: props.core.y,
+        backgroundColor: props.core.backgroundColor,
+      }}
+    />
+  );
+}
+
+  
+
+var ents = {knight: { core: new Knight(), renderer: <CreateBrick />},};
```

Comment:
You should declare global variables at the top, either with a default value or just as a declaration, e.g.:
```
// Store a global reference to the game engine dictionary
var ents
```
And you should move the code that creates the Knight into `export default function Dodge()`.